### PR TITLE
Purge stdenv.cross

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

`stdenv.cross` is vague: better to use `buildPlatform`, `hostPlatform` or `targetPlatform`. This is a rather tedious task so anyone that sees this, feel free to push to this PR (if you are a contributor) or send a PR against the PR (if you are not).

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @bjornfor @matthewbauer @Dridus

### CODE REVIEW

File: pkgs/development/libraries/libxml2/default.nix [hunk 1]
```diff
@@ -1,7 +1,9 @@
 { stdenv, lib, fetchurl, fetchpatch
 , zlib, xz, python2, findXMLCatalogs, libiconv
-, pythonSupport ? (! stdenv ? cross)
+, buildPlatform, hostPlatform
+, pythonSupport ? buildPlatform == hostPlatform
 , icuSupport ? false, icu ? null }:
+}:
```
Comment (Mic92) [inline pkgs/development/libraries/libxml2/default.nix]: closes parentheses twice.

Comment (Ericson2314) [inline pkgs/development/libraries/libxml2/default.nix]: Thanks; fixed

**Review by 0xABAB — APPROVED** [review]

**Comment by Ericson2314** [comment]
@bjornfor that's my fault with the sed before the manual fix up. I hope there's a way to filter-branch to fix that....

**Comment by bjornfor** [comment]
@Ericson2314. No problem. I think you can script the git filter-branch thing.

**Pull request merged**